Content.Server/Blob/BlobObserverSystem.cs

```diff
@@ -0,0 +1,862 @@
+                blobNodes.Add(ent);
+        }
+
+        if (blobNodes.Count == 0)
+        {
+            _popup.PopupEntity(Loc.GetString("blob-not-have-nodes"), uid, uid, PopupType.Large);
+            args.Handled = true;
+            return;
+        }
+
+        _transform.SetCoordinates(uid, Transform(_random.Pick(blobNodes)).Coordinates);
+        args.Handled = true;
+    }
+
+    private void OnCreateBlobbernaut(EntityUid uid, BlobObserverComponent observerComponent,
+        BlobCreateBlobbernautActionEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (observerComponent.Core == null ||
+            !TryComp<BlobCoreComponent>(observerComponent.Core.Value, out var blobCoreComponent))
+            return;
+
+        var gridUid = args.Target.GetGridUid(EntityManager);
+
+        if (!_map.TryGetGrid(gridUid, out var grid))
+        {
+            return;
+        }
+
+        var centerTile = grid.GetLocalTilesIntersecting(
+            new Box2(args.Target.Position, args.Target.Position)).ToArray();
+
+        EntityUid? blobTile = null;
+
+        foreach (var tileRef in centerTile)
+        {
+            foreach (var ent in grid.GetAnchoredEntities(tileRef.GridIndices))
+            {
+                if (!HasComp<BlobFactoryComponent>(ent))
+                    continue;
+                blobTile = ent;
+                break;
+            }
+        }
+
+        if (blobTile == null || !TryComp<BlobFactoryComponent>(blobTile, out var blobFactoryComponent))
+        {
+            _popup.PopupEntity(Loc.GetString("blob-target-factory-blob-invalid"), uid, uid, PopupType.LargeCaution);
+            return;
+        }
+
+        if (blobFactoryComponent.Blobbernaut != null)
+        {
+            _popup.PopupEntity(Loc.GetString("blob-target-already-produce-blobbernaut"), uid, uid, PopupType.LargeCaution);
+            return;
+        }
+
+        if (!_blobCoreSystem.TryUseAbility(uid, observerComponent.Core.Value, blobCoreComponent, blobCoreComponent.BlobbernautCost))
+            return;
+
+        var ev = new ProduceBlobbernautEvent();
+        RaiseLocalEvent(blobTile.Value, ev);
+
+        _popup.PopupEntity(Loc.GetString("blob-spent-resource", ("point", blobCoreComponent.BlobbernautCost)),
+            blobTile.Value,
+            uid,
+            PopupType.LargeCaution);
+
+        args.Handled = true;
+    }
+
+    private void OnBlobToCore(EntityUid uid, BlobObserverComponent observerComponent,
+        BlobToCoreActionEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (observerComponent.Core == null ||
+            !TryComp<BlobCoreComponent>(observerComponent.Core.Value, out var blobCoreComponent))
+            return;
+
+        _transform.SetCoordinates(uid, Transform(observerComponent.Core.Value).Coordinates);
+    }
+
+    private void OnCreateNode(EntityUid uid, BlobObserverComponent observerComponent,
```

Comment:
A large chunk of the logic in here is core to blob functioning and shouldn't be part of the player interface.